piker/brokers/kraken.py

```diff
@@ -313,29 +305,41 @@ async def submit_limit(
         price: float,
         action: str,
         size: float,
-        reqid: int = None,
-    ) -> int:
+        reqid: str = None,
+    ) -> dict:
         '''
         Place an order and return integer request id provided by client.
 
         '''
-        # Build order data for kraken api
-        data = {
-            "userref": reqid,
-            "ordertype": "limit",
-            "type": action,
-            "volume": str(size),
-            "pair": symbol,
-            "price": str(price),
-            # set to True test AddOrder call without a real submission
-            "validate": False
-        }
-        return await self.endpoint('AddOrder', data)
+        if reqid is None:
+            # Build order data for kraken api
+            data = {
```

Comment:
the `data: dict` duplication isn't necessary right?

in the edit case can't you just update with the `txid` ?

Reply:
you mean, create the `dict` with the common keys from both end points, and then depending on which one we are using, add the extras?

Reply:
yes

Reply:
aight